### DISCUSSION TITLE
How to save a Vinyl object to filesystem?

### DISCUSSION THREAD

**Discussion by rossjrw** [gulpjs/gulp#2525] [Help] [answered]
[Documentation for Vinyl objects](https://gulpjs.com/docs/en/api/vinyl) states the following:

> They may also be persisted to the file system using `dest()`.

That doesn't seem to be the case for raw Vinyl objects - they don't implement a `dest` method and the `dest` function that Gulp exports doesn't take an argument, being designed for use in `pipe`, which Vinyl objects don't implement either.

How do I persist a Vinyl object created with `new Vinyl()` to the filesystem?

**Comment by yocontra** [comment, marked as answer]
@rossjrw You use an adapter like vinyl-fs - there are many adapters that output or accept vinyl files as input, `vinyl-fs` (which uses your local filesystem) is one of them.

**Reply by rossjrw** [reply]
Thank you for your input - I ended up bypassing Vinyl altogether and simply using `fs.writeFile` on the string that would have been passed to the Vinyl constructor. Using Vinyl in this case was an overcomplication, as the file being written didn't need to go through Gulp simply because Gulp was being used in the project.

**Reply by yocontra** [reply]
@rossjrw Yep absolutely - use the easiest tool for the job, you don't need to use any of the vinyl stuff to write a basic text string to the file system.
